### PR TITLE
MGDAPI-4375 / Added CriticalMetricsMissing Alert

### PR DESCRIPTION
**Pull request by den-rgb**
# Issue link
[MGDAPI-4375](https://issues.redhat.com/browse/MGDAPI-4375)

# What
Added an alert as well as a SOP to check if Critical metrics are missing.

# Verification steps
1. Checkout this branch.
2. Install rhoam through OLM.
3. Make sure alert is not triggered during install.
4. Once installed is finished disable rhmi-operator pod and wait for alert to trigger.
5. Enable the pod again.
6. Upgrade the version.
7. Make sure alert doesn't trigger.

- You can use images: 
quay.io/dmoskale/managed-api-service:v1.26.3
quay.io/dmoskale/managed-api-service:v1.26.4

- index: 
quay.io/dmoskale/managed-api-service-index:1.26.3
quay.io/dmoskale/managed-api-service-index:1.26.4

### CODE REVIEW

**Comment by kryanbeane** [comment]
/lgtm
Alert fired when RHMI operaotr pod was scaled down, and didn't fire during upgrade or install

File: pkg/products/observability/prometheusRules.go [hunk 1]
```diff
@@ -341,6 +341,23 @@ func (r *Reconciler) newAlertsReconciler(logger l.Logger, installType string) re
...
+			Namespace: namespace,
+			GroupName: fmt.Sprintf("%s-general.rules", installationName),
+			Rules: []monitoringv1.Rule{
+				{
+					Alert: fmt.Sprintf("%sCriticalMetricsMissing", strings.ToUpper(installationName)),
+					Annotations: map[string]string{
+						"sop_url": resources.SopUrlCriticalMetricsMissing,
+						"message": fmt.Sprintf("%s metrics have been missing for more than 5 minutes", strings.ToUpper(installationName)),
+					},
+					Expr:   intstr.FromString(`(` + installationName + `_version{version=~".*", to_version=~""} > 0) * on(pod) group_left(` + installationName + `_custom_domain)(absent(` + installationName + `_custom_domain)) or absent(` + installationName + `_version)`),
```
Comment (MStokluska) [inline pkg/products/observability/prometheusRules.go:355]: is the empty space before the installationName necessary?

Comment (MStokluska) [reply pkg/products/observability/prometheusRules.go:355]: nvm! :man_facepalming:

Comment (MStokluska) [reply pkg/products/observability/prometheusRules.go:355]: Although I would keep it the same as all other alerts; using the fmt.Sprintf if you can like here:
https://github.com/integr8ly/integreatly-operator/pull/2802/files#diff-7139745d8ee64821c1d2176216e15673c888f7cd064df149a05527f34027efd4R81 to keep consistent

Comment (den-rgb) [reply pkg/products/observability/prometheusRules.go:355]: I don't mind changing it but most of the alerts above it don't use the fmt.Sprintf in the experession

Comment (MStokluska) [reply pkg/products/observability/prometheusRules.go:355]: @den-rgb most of them (if not all) do use fmt.Sprintf if we need to pass in a variable to the query. I'd like this alert to stay consistent with others if possible.

Comment (MStokluska) [reply pkg/products/observability/prometheusRules.go:355]: also, looking at only the file you are changing, notice how the other alerts are using it; it is not used when we don't need to take in a variable

Comment (MStokluska) [reply pkg/products/observability/prometheusRules.go:355]: and just to re-iterate; this is not a big deal and the only reason I've picked that up is that I made a mistake when reading it for the first time, due to it being constructed in a different way than others.

**Comment by kryanbeane** [comment]
@den-rgb will leave lgtm again since the change was minor and doesn't need rereview, but can you fix the new commit message to have `MGDAPI-4375`

**Comment by kryanbeane** [comment]
/lgtm
since last change was minor formatting edit, shouldn't require reverification

**Pull request closed without merging**